packages/tenant-process/src/services/tenantService.ts

```diff
@@ -261,6 +286,76 @@ export function tenantServiceBuilder(
       }
     },
 
+    async addCertifiedAttribute(
+      {
+        tenantId,
+        tenantAttributeSeed,
+        organizationId,
+        correlationId,
+      }: {
+        tenantId: TenantId;
+        tenantAttributeSeed: tenantApi.CertifiedTenantAttributeSeed;
+        organizationId: TenantId;
+        correlationId: string;
+      },
+      logger: Logger
+    ): Promise<Tenant> {
+      logger.info(
+        `Add certified attribute ${tenantAttributeSeed.id} to tenant ${tenantId}`
+      );
+
+      const requesterTenant = await retrieveTenant(
+        organizationId,
+        readModelService
+      );
+
+      const certifierId = getTenantCertifierId(requesterTenant.data);
+
+      const attribute = await retrieveAttribute(
+        unsafeBrandId(tenantAttributeSeed.id),
+        readModelService
+      );
+
+      if (attribute.kind !== attributeKind.certified) {
+        throw attributeNotFound(attribute.id);
+      }
+
+      if (!attribute.origin || attribute.origin !== certifierId) {
+        throw attributeDoesNotBelongToCertifier(
+          attribute.id,
+          organizationId,
+          tenantId
+        );
+      }
+
+      const targetTenant = await retrieveTenant(tenantId, readModelService);
+
+      // eslint-disable-next-line functional/no-let
+      let updatedTenant = await assignCertifiedAttribute({
+        targetTenant: targetTenant.data,
+        attribute,
+      });
+
+      updatedTenant = await revaluateTenantKind({
+        tenant: updatedTenant,
+        version: targetTenant.metadata.version,
+        readModelService,
+        dbInstance,
+        correlationId,
+      });
+
+      await repository.createEvent(
+        toCreateEventTenantCertifiedAttributeAssigned(
+          targetTenant.metadata.version + 1,
```

Comment:
2 more points:
- this `+ 1` must be done only if there is an event to update the kind (so the kind has actually been changed)
- `TenantCertifiedAttributeAssigned` should be saved before `TenantKindUpdated` (logically, first the attribute is assigned, and then the kind is updated)